backend/Watchdog.AspNetCore/WatchdogClientProvider.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Microsoft.AspNetCore.Http;
+
+namespace Watchdog.AspNetCore
+{
+    public interface IWatchdogAspNetCoreClientProvider
+    {
```

Comment:
The DefaultWatchdogAspNetCoreClientProvider class that implements this interface has virtual methods that you can simply override instead of creating a new class that will implement this interface again. This class and interface are a holistic layer of abstraction, which is the reason why I put them in one file.

Reply:
Look at the answer above.